File: PG_ELLAv2.2/DlogPiDThetaREINFORCE.m
function [der,der2] = DlogPiDThetaREINFORCE(policy,x,u,param)

% Programmed by Jan Peters (jrpeters@usc.edu).
N = param.param.N;
M = param.param.M;

sigma = max(policy.sigma,0.00001);
k = policy.theta;
xx = x;
der = [];
for i=1:M
    der = [der;(u(i)-k(1+N*(i-1):N*(i-1)+N)'*xx)*xx/(sigma(i)^2)];
end

File: src/vertical_control/scripts/pg_ella/DlogPiDThetaREINFORCE.py
# -*- coding: utf-8 -*-
"""
Created on Tue Aug 11 12:35:20 2015

@author: robot3
"""

import numpy as np


def DlogPiDThetaREINFORCE(policy, x, u, param):
    N = param.param.N
    M = param.param.M

    sigma = np.max(policy.sigma, 0.00001)
    k = policy.theta

    der = np.empty(shape=(0, N))
    for i in range(M):
        der = np.concatenate((der, np.dot((u[i]-np.dot(k[N*(i):N*(i)+N].conj().T, x)), x.conj().T) / (sigma[i]**2)))

    return der